botnim/benchmark/assistant_loop.py

```diff
@@ -64,34 +94,128 @@ def assistant_loop(client: OpenAI, assistant_id, question=None, thread=None, not
-                        for result in(tool_call.file_search.results or []):
+                        for result in (tool_call.file_search.results or []):
                             text = result.content[0].text if result.content else None
                             if text:
                                 notes.append(f'>>\n{text}\n<<')
 
-        if run.status == 'completed': 
+        if run.status == 'completed':
+            # Log assistant's response when run is completed
+            messages = client.beta.threads.messages.list(thread_id=thread.id, order='desc', limit=1)
+            for message in messages:
+                if message.role == "assistant":
+                    with open(log_file, 'a', encoding='utf-8') as f:
+                        f.write("\nAssistant Response:\n")
+                        for content in message.content:
+                            if content.type == 'text':
+                                f.write(f"{content.text.value}\n")
+                                break  # Only write the latest response
+            
+            # Log completion
+            with open(log_file, 'a', encoding='utf-8') as f:
+                f.write("\n=== Run Completed ===\n\n")
             break
 
         for tool in run.required_action.submit_tool_outputs.tool_calls:
             arguments = json.loads(tool.function.arguments)
-            if tool.function.name == 'DatasetDBQuery':
-                arguments['page_size'] = 30
-            output = get_openapi_output(openapi_spec, tool.function.name, arguments)
-            if tool.function.name == 'DatasetInfo':
-                output = get_dataset_info_cache(arguments, output)
-            tool_outputs.append(dict(
-                tool_call_id=tool.id,
-                output=json.dumps(output, ensure_ascii=False, indent=2)
-            ))
+            output = None  # Initialize output variable
+            
+            # Log tool input
+            with open(log_file, 'a', encoding='utf-8') as f:
+                f.write(f"\nTool Input:\n  Tool ID: {tool.id}\n  Name: {tool.function.name}\n  Arguments:\n")
+                for key, value in arguments.items():
+                    f.write(f"    {key}: {value}\n")
+            
+            # Handle different tool types
+            if tool.function.name.startswith('search_'):
+                # Handle the search_takanon__context__dev pattern
+                # Remove 'search_' prefix and '__dev' suffix if present
+                tool_name = tool.function.name[len('search_'):]
+                if tool_name.endswith('__dev'):
+                    tool_name = tool_name[:-len('__dev')]
+                
+                # Split into bot_name and context_name
+                parts = tool_name.split('__', 1)
+                bot_name = parts[0]
+                context_name = parts[1] if len(parts) > 1 else ''
+                
+                # Get num_results from arguments or use default
```

Comment:
don't we want to use the default number in the context?

Reply:
right. fixed.